### PR TITLE
WT-6430 Move WT_CONN_SERVER flags into their own field (v4.2 backport)

### PR DESCRIPTION
**Pull request by etienneptl**

### CODE REVIEW

File: src/conn/conn_open.c [hunk 1]
```diff
@@ -78,7 +78,7 @@ __wt_connection_close(WT_CONNECTION_IMPL *conn)
      * wiredtiger_open hits an error (as well as during normal shutdown). Assert they're not
      * running.
      */
-    WT_ASSERT(session, !F_ISSET(conn, WT_CONN_SERVER_ASYNC | WT_CONN_SERVER_LSM));
+    WT_ASSERT(session, !FLD_ISSET(conn->server_flags, WT_CONN_SERVER_ASYNC | WT_CONN_SERVER_LSM));
```
Comment (etienneptl) [inline src/conn/conn_open.c:81]: this '|' is not present in the develop branch, see https://github.com/wiredtiger/wiredtiger/pull/6264/files
Should we leave it or align it with the develop branch ?

**Comment by etienneptl** [comment]
retest this please

File: src/conn/conn_open.c (hunk 1 above)
Comment (luke-pearson) [reply src/conn/conn_open.c:81]: Leave it as is, the flag `WT_CONN_SERVER_ASYNC` was removed in WT-6674 which hasn't been backported to 4.2.

File: src/include/connection.h [hunk 2]
```diff
@@ -515,39 +515,46 @@ struct __wt_connection_impl {
      */
     WT_FILE_SYSTEM *file_system;
 
+/*
+ * Server subsystem flags.
+ */
+/* AUTOMATIC FLAG VALUE GENERATION START */
+#define WT_CONN_SERVER_CAPACITY 0x01u
```
Comment (luke-pearson) [inline src/include/connection.h:522]: I think you need `WT_CONN_SERVER_ASYNC` here?

Comment (etienneptl) [reply src/include/connection.h:522]: At the end of the list line 556

Comment (luke-pearson) [reply src/include/connection.h:522]: Yeah but if the idea behind this change is to move `WT_CONN_SERVER` flags to their own field then `WT_CONN_SERVER_ASYNC` should get moved too.

File: src/conn/conn_open.c (hunk 1 above)
Comment (sueloverso) [reply src/conn/conn_open.c:81]: That also means you need to change all uses of `WT_CONN_SERVER_ASYNC` in the `src/async` directory.

**Review by sueloverso — CHANGES_REQUESTED** [review]
There are changes needed for async that exists in 4.2 but does not exist in develop.

**Review by sueloverso — APPROVED** [review]
LGTM. It was definitely more than just a cherry pick.

**Review by luke-pearson — APPROVED** [review]
LGTM.

**Pull request merged**